src/architecture.c

```diff
@@ -322,37 +322,121 @@ static void arithm(arch_core* core)
 
 static void mov_imm(arch_core* core)
 {
+	arch_uint eff_addr;
+	if (((cond_imm_data)core->regs.ir.data).dreg == 0)
+	{
+		eff_addr = ((cond_imm_data)core->regs.ir.data).addr;
+	}
+	else
+	{
+		eff_addr = ((cond_imm_data)core->regs.ir.data).dreg;
+	}
 
+	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg;
+	arch_word* reg = help_get_reg(core, eff_reg);
+	eff_addr = *help_get_reg(core, eff_addr);
```

Comment:
Remember what we discussed on discord about this implementation:
```c
static void mov_imm(arch_core* core)
{
  arch_uint eff_data = ((cond_imm_data)core->regs.ir.data).addr;
  arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg;
  *help_get_reg(core, eff_reg) = eff_data;
}
```

and this condition in `decode()`:
```c
case MOVI:
  if (((cond_imm_data)core->regs.ir.data).dreg != 0)
  {
     core->regs.pc = INTRPT_OFF_UNDEF;
     core->pipeline[CORE_EXE_STEP] = interrupt;
  }
```
Try to do something similar to this with the other functions.

Reply:
Okay I updated this gonna make sure the decode() part works, IIRC you said that for each if the dreg does not equal 0 then it should trigger an interrupt?